stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -0,0 +1,292 @@
+                    ("route_id.flow_id", "=", False),
+                ]
+            )
+            flow.impacted_route_ids = rules.route_id
+
+    @api.constrains(
+        "warehouse_id", "from_picking_type_id", "carrier_ids", "move_domain"
+    )
+    def _constrains_uniq(self):
+        args = [
+            ("warehouse_id", "=", self.warehouse_id.id),
+            ("from_picking_type_id", "=", self.from_picking_type_id.id),
+        ]
+        flows = self.search(args) - self
+        for flow in flows:
+            if (
+                self.carrier_ids & flow.carrier_ids
+                or not self.carrier_ids
+                and not flow.carrier_ids
+            ) and self.move_domain == flow.move_domain:
+                raise UserError(
+                    _("Existing flow '%s' already applies on these kind of moves.")
+                    % flow.name
+                )
+
+    @api.onchange("name")
+    def onchange_name(self):
+        self.sequence_prefix = "".join(
+            [w[0] for w in (self.name or "").split()]
+        ).upper()
+
+    @api.onchange("warehouse_id")
+    def onchange_warehouse_id(self):
+        self.from_picking_type_id = self.warehouse_id.out_type_id
+
+    def _generate_delivery_route(self):
+        self.ensure_one()
+        if self.delivery_route_id:
+            return False
+        # Re-generate the WH configuration with the chosen delivery steps in a
+        # savepoint to get a whole route+rules+picking types configured almost
+        # automatically (to not re-invent the wheel).
+        # Delivery route values are then copied to generate a new one.
+        # Copy some picking types with new sublocations first so the new
+        # delivery route will use them.
+        pick_type, pack_type, out_type = False, False, False
+        out_type = self.warehouse_id.out_type_id.copy(
+            {
+                "name": f"{self.warehouse_id.out_type_id.name} {self.sequence_prefix}",
+            }
+        )
+        out_type.sequence_id.prefix += f"{self.sequence_prefix}/"
+        output_loc, pack_loc = False, False
+        if "pick_" in self.delivery_steps:
+            pick_type = self.warehouse_id.pick_type_id.copy(
+                {
+                    "name": f"{self.warehouse_id.pick_type_id.name} {self.sequence_prefix}",
+                }
+            )
+            pick_type.sequence_id.prefix += f"{self.sequence_prefix}/"
+            # Create a dedicated 'Output/SUB' location
+            output_loc = self.warehouse_id.wh_output_stock_loc_id.copy(
+                {
+                    "name": self.sequence_prefix,
+                    "location_id": self.warehouse_id.wh_output_stock_loc_id.id,
+                }
+            )
+            pick_type.default_location_dest_id = output_loc
+            out_type.default_location_src_id = output_loc
+            if "pack_" in self.delivery_steps:
+                pack_type = self.warehouse_id.pack_type_id.copy(
```

Comment:
The warehouse picking type could be archived. When you copy, you need to ensure it is not archived anymore